vcf.c

```diff
@@ -818,6 +819,14 @@ int bcf_hdr_parse(bcf_hdr_t *hdr, char *htxt)
             }
             p += len;
         }
+        if (hrec == NULL && len != 0) {
+            char *eol = strchr(p, '\n');
+            if (*p != '\0') {
+                hts_log_warning("Could not parse header line: %.*s",
+                                eol ? (int)(eol - p) : INT_MAX, p);
+            }
+            return -1;
```

Comment:
I just noticed that this changes the behaviour described in the [comment](https://github.com/samtools/htslib/blob/d99bf404/vcf.c#L831) below.  Does this mean the future has arrived?  I fear we may get complaints about suddenly unreadable files again.

Would it be better to only give up on the out-of-memory case (now indicated by `len == -1`) and just skip the bad lines as before?

Reply:
Ugh, so it's deliberate that it spams a million errors and ignores all of them?

*splutter*

I guess we could revert that bit then if needed.  I think it was in its own commit anyway.

Reply:
Apparently so.  I will make the necessary adjustments.

Reply:
Thanks